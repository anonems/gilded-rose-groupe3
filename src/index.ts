export class Item {
  name: string;
  sellIn: number;
  quality: number;

  constructor(name, sellIn, quality) {
    this.name = name;
    this.sellIn = sellIn;
    this.quality = quality;
  }
}

export class GildedRose {
  items: Array<Item>;

  constructor(items = [] as Array<Item>) {
    this.items = items;
  }

<<<<<<< Updated upstream
  updateQuality() {
    for (let i = 0; i < this.items.length; i++) {
      if (this.items[i].name != "Aged Brie" && this.items[i].name != "Backstage passes to a TAFKAL80ETC concert") {
        if (this.items[i].quality > 0) {
          if (this.items[i].name != "Sulfuras, Hand of Ragnaros") {
            this.items[i].quality = this.items[i].quality - 1;
          }
        }
      } else {
        if (this.items[i].quality < 50) {
          this.items[i].quality = this.items[i].quality + 1;
          if (this.items[i].name == "Backstage passes to a TAFKAL80ETC concert") {
            if (this.items[i].sellIn < 11) {
              if (this.items[i].quality < 50) {
                this.items[i].quality = this.items[i].quality + 1;
              }
            }
            if (this.items[i].sellIn < 6) {
              if (this.items[i].quality < 50) {
                this.items[i].quality = this.items[i].quality + 1;
              }
            }
          }
        }
      }
      if (this.items[i].name != "Sulfuras, Hand of Ragnaros") {
        this.items[i].sellIn = this.items[i].sellIn - 1;
      }
      if (this.items[i].sellIn < 0) {
        if (this.items[i].name != "Aged Brie") {
          if (this.items[i].name != "Backstage passes to a TAFKAL80ETC concert") {
            if (this.items[i].quality > 0) {
              if (this.items[i].name != "Sulfuras, Hand of Ragnaros") {
                this.items[i].quality = this.items[i].quality - 1;
              }
            }
          } else {
            this.items[i].quality = this.items[i].quality - this.items[i].quality;
          }
        } else {
          if (this.items[i].quality < 50) {
            this.items[i].quality = this.items[i].quality + 1;
          }
        }
      }
=======
  private updateAgedBrie(item) {
    if (item.quality < 50) {
      item.quality++;
>>>>>>> Stashed changes
    }

<<<<<<< Updated upstream
    return this.items;
  }
}
=======
  private updateBackstage(item) {
    if (item.quality < 50) {
      item.quality++;
      if (item.sellIn < 11 && item.quality < 50) {
        item.quality++;
      }
      if (item.sellIn < 6 && item.quality < 50) {
        item.quality++;
      }
    }
    if (item.sellIn < 0) {
      item.quality = 0;
    }
  }

  private update(item) {
    if (item.name !== "Sulfuras, Hand of Ragnaros") {
      if (item.quality > 0) {
        item.quality--;
      }
      if (item.sellIn < 0 && item.quality > 0) {
        item.quality--;
      }
    }
  }

  updateQuality() {
    this.items.forEach(item => {
      switch (item.name) {
        case "Aged Brie":
          this.updateAgedBrie(item);
          break;
        case "Backstage passes to a TAFKAL80ETC concert":
          this.updateBackstage(item);
          break;
        default:
          this.update(item);
      }
    })
    return this.items;
  }

}
>>>>>>> Stashed changes
