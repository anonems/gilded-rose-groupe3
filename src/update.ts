export class Item {
    name: string; //intitulé de l'article
    sellIn: number; //nombre de jours max pour vendre un article
    quality: number; //valeur de l'objet (selon sa qualité) toujours positive et inferieur a 50. (>=50)

    constructor(name, sellIn, quality) {
        this.name = name;
        this.sellIn = sellIn;
        this.quality = quality;
    }
}

export class GildedRose {
    items: Array<Item>;

    constructor(items = [] as Array<Item>) {
        this.items = items;
    }

    private updateAgedBrie(item) {
        if (item.quality < 50) {
            item.quality++;
        }
    }

    private updateBackstage(item) {
        if (item.quality < 50) {
            item.quality++;
            if (item.sellIn < 11 && item.quality < 50) {
                item.quality++;
            }
            if (item.sellIn < 6 && item.quality < 50) {
                item.quality++;
            }
        }
        if (item.sellIn < 0) {
            item.quality = 0;
        }
    }

    private update(item) {
        if (item.name !== "Sulfuras, Hand of Ragnaros") {
            if (item.quality > 0) {
                item.quality--;
            }
            if (item.sellIn < 0 && item.quality > 0) {
                item.quality--;
            }
        }
    }

    updateQuality() {
        this.items.forEach(item => {
            // logs to display the state avant update de la qualité
            console.log(`Item: ${item.name}, SellIn: ${item.sellIn}, Quality: ${item.quality}`);

            switch (item.name) {
                case "Aged Brie":
                    this.updateAgedBrie(item);
                    break;
                case "Backstage passes to a TAFKAL80ETC concert":
                    this.updateBackstage(item);
                    break;
                default:
                    this.update(item);
            }

            // logs to display the state après update de la qualité
            console.log(`Updated - Item: ${item.name}, SellIn: ${item.sellIn}, Quality: ${item.quality}`);
        });

        return this.items;
    }
}
