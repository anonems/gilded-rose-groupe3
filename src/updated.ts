export class Item {
  name: string;
  sellIn: number;
  quality: number;

  constructor(name, sellIn, quality) {
    this.name = name;
    this.sellIn = sellIn;
    this.quality = quality;
  }
}

export class GildedRose {
  items: Array<Item>;

  constructor(items = [] as Array<Item>) {
    this.items = items;
  }

  updateQuality() {
    for (let i = 0; i < this.items.length; i++) {
      const item = this.items[i];

      // logs pour afficher l'état avant update de la qualité
      console.log(`Item: ${item.name}, SellIn: ${item.sellIn}, Quality: ${item.quality}`);

      if (item.name === "Aged Brie") {
        if (item.quality < 50) {
          item.quality++;
        }
      } else if (item.name === "Backstage passes to a TAFKAL80ETC concert") {
        if (item.quality < 50) {
          item.quality++;
          if (item.sellIn < 11 && item.quality < 50) {
            item.quality++;
          }
          if (item.sellIn < 6 && item.quality < 50) {
            item.quality++;
          }
        }
        if (item.sellIn < 0) {
          item.quality = 0;
        }
      } else if (item.name !== "Sulfuras, Hand of Ragnaros") {
        if (item.quality > 0) {
          item.quality--;
        }
        if (item.sellIn < 0 && item.quality > 0) {
          item.quality--;
        }
      }

      // logs pour afficher l'état après update de la qualité
      console.log(`Updated - Item: ${item.name}, SellIn: ${item.sellIn}, Quality: ${item.quality}`);

      if (item.name !== "Sulfuras, Hand of Ragnaros") {
        item.sellIn--;
      }
    }

    return this.items;
  }
}
