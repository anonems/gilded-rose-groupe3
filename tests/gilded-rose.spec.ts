import { GildedRose, Item } from '../src/updated'; // Assurez-vous que l'importation de vos classes est correcte.

describe('Gilded Rose', () => {
  it('should match snapshot for Conjured item', () => {
    const gildedRose = new GildedRose([new Item('Conjured', 2, 10)]);
    gildedRose.updateQuality();
    expect(gildedRose.items).toMatchSnapshot();
  });

  it('should match snapshot for Aged Brie with sellIn = 0', () => {
    const gildedRose = new GildedRose([new Item('Aged Brie', 0, 10)]);
    gildedRose.updateQuality();
    expect(gildedRose.items).toMatchSnapshot();
  });

  it('should match snapshot for Backstage passes with sellIn = 0', () => {
    const gildedRose = new GildedRose([new Item('Backstage passes to a TAFKAL80ETC concert', 0, 10)]);
    gildedRose.updateQuality();
    expect(gildedRose.items).toMatchSnapshot();
  });

  it('should match snapshot for Sulfuras with sellIn = 0', () => {
    const gildedRose = new GildedRose([new Item('Sulfuras, Hand of Ragnaros', 0, 80)]);
    gildedRose.updateQuality();
    expect(gildedRose.items).toMatchSnapshot();
  });

  it('should match snapshot for empty items array', () => {
    const gildedRose = new GildedRose([]);
    gildedRose.updateQuality();
    expect(gildedRose.items).toMatchSnapshot();
  });

  it('should match snapshot for normal item with sellIn > 0 and quality > 0', () => {
    const gildedRose = new GildedRose([new Item('Normal Item', 5, 10)]);
    gildedRose.updateQuality();
    expect(gildedRose.items).toMatchSnapshot();
  });

  it('should match snapshot for normal item with sellIn = 0 and quality > 0', () => {
    const gildedRose = new GildedRose([new Item('Normal Item', 0, 10)]);
    gildedRose.updateQuality();
    expect(gildedRose.items).toMatchSnapshot();
  });

  it('should match snapshot for normal item with sellIn > 0 and quality = 0', () => {
    const gildedRose = new GildedRose([new Item('Normal Item', 5, 0)]);
    gildedRose.updateQuality();
    expect(gildedRose.items).toMatchSnapshot();
  });

  it('should match snapshot for normal item with sellIn = 0 and quality = 0', () => {
    const gildedRose = new GildedRose([new Item('Normal Item', 0, 0)]);
    gildedRose.updateQuality();
    expect(gildedRose.items).toMatchSnapshot();
  });

});
