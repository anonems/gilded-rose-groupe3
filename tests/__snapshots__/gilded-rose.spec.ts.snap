// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gilded Rose should match snapshot for Aged Brie with sellIn = 0 1`] = `
[
  Item {
    "name": "Aged Brie",
    "quality": 11,
    "sellIn": -1,
  },
]
`;

exports[`Gilded Rose should match snapshot for Backstage passes with sellIn = 0 1`] = `
[
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 13,
    "sellIn": -1,
  },
]
`;

exports[`Gilded Rose should match snapshot for Conjured item 1`] = `
[
  Item {
    "name": "Conjured",
    "quality": 9,
    "sellIn": 1,
  },
]
`;

exports[`Gilded Rose should match snapshot for Sulfuras with sellIn = 0 1`] = `
[
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
]
`;

exports[`Gilded Rose should match snapshot for empty items array 1`] = `[]`;

exports[`Gilded Rose should match snapshot for normal item with sellIn = 0 and quality = 0 1`] = `
[
  Item {
    "name": "Normal Item",
    "quality": 0,
    "sellIn": -1,
  },
]
`;

exports[`Gilded Rose should match snapshot for normal item with sellIn = 0 and quality > 0 1`] = `
[
  Item {
    "name": "Normal Item",
    "quality": 9,
    "sellIn": -1,
  },
]
`;

exports[`Gilded Rose should match snapshot for normal item with sellIn > 0 and quality = 0 1`] = `
[
  Item {
    "name": "Normal Item",
    "quality": 0,
    "sellIn": 4,
  },
]
`;

exports[`Gilded Rose should match snapshot for normal item with sellIn > 0 and quality > 0 1`] = `
[
  Item {
    "name": "Normal Item",
    "quality": 9,
    "sellIn": 4,
  },
]
`;
